dts/arm/st/wl/stm32wl.dtsi

```diff
@@ -219,6 +264,108 @@
 			status = "disabled";
 			label = "SPI_2";
 		};
+
+		subghzspi: spi@58010000 {
+			compatible = "st,stm32-spi-fifo", "st,stm32-spi";
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0x58010000 0x400>;
+			interrupts = <44 5>;
```

Comment:
Can you check the position of the interrupt as the entry 44 is mentioned as "Reserved" by the RefMan
subghzspi is present at position 30 of the CPU2 for the stm32wl5x.

Reply:
Hey you are right! I was cross referencing stm32wle5xx.h from the cube package when I did this, the header file defines:

```
#if defined(CORE_CM0PLUS)
...
SUBGHZSPI_IRQn = 30, 
...
#else
...
SUBGHZSPI_IRQn = 44,
...
```

and stm32wle5xx.h only defines it on 44. Curiously enough there is a Radio IRQ on channel 44 of the EXTI, but it's not the SPI Interrupt. And the SPI driver in the cube package uses that SPI in polling mode anyway.

But setting this to 30 would not be correct either since it would try to resolve to the wrong nvic controller, and we don't have anything for the second cpu defined in the dts yet. Plus it would not help me because I'm working on a STM32WLE5, the single core version. :-)

Do you have a contact to check if maybe the definition in the cube file is correct and that reserved line is actually usable and the reference manual has to be rectified? Otherwise it means that this SPI cannot be used in interrupt mode in a single core setup.

Reply:
For what is worth, I hacked together a sample to probe something on that SPI and the interrupt does fire:

```
Breakpoint 1, spi_stm32_isr (dev=0x20000160 <__device_dts_ord_165>)
    at zephyrproject/zephyr/drivers/spi/spi_ll_stm32.c:409
409             struct spi_stm32_data *data = dev->data;
(gdb) p *dev
$1 = {name = 0x8006649 "SUBGHZSPI", config = 0x8005d74 <spi_stm32_cfg_0>, api = 0x8005d4c <api_funcs>, 
  state = 0x200004cc <__devstate_dts_ord_165>, data = 0x20000020 <spi_stm32_dev_data_0>, 
  handles = 0x8005b88 <__devicehdl_DT_N_S_soc_S_spi_58010000>}
```

Found few other missing bits on the clock setup to get to that point though so I'm adding a patch for those as well. :-)

Reply:
This is probably a missing line in the ref manual (of the stm32wleX - RM0461): The stm32wle5xx.h from the cube is setting the SUBGHZSPI_IRQn at position = 44,  for the M4. Let me check internally.

Reply:
Right, sorry I pasted the wrong file name on my first reply, stm32wl55xx.h defines that on both 30 and 44 depending on whether it's building for the M0 or M4, and stm32wle5xx.h only has it on 44, and only has the M4.  Thanks for checking it!

Reply:
After checking, the entry 44  of the NVIC  is the SUBGHZ SPI instance.